internal/data/generated_trace_test.go

```diff
@@ -51,6 +176,81 @@ func TestSpanEvent(t *testing.T) {
 	assert.EqualValues(t, generateTestSpanEvent(), ms)
 }
 
+func TestSpanLinkSlice(t *testing.T) {
+	es := NewSpanLinkSlice(0)
+	assert.EqualValues(t, 0, es.Len())
+	es = newSpanLinkSlice(&[]*otlptrace.Span_Link{})
+	assert.EqualValues(t, 0, es.Len())
+	es = NewSpanLinkSlice(13)
+	defaultVal := NewSpanLink()
+	testVal := generateTestSpanLink()
+	assert.EqualValues(t, 13, es.Len())
+	for i := 0; i < es.Len(); i++ {
+		assert.EqualValues(t, defaultVal, es.Get(i))
+		fillTestSpanLink(es.Get(i))
+		assert.EqualValues(t, testVal, es.Get(i))
+	}
+
+	// Test resize.
+	const resizeLo = 2
+	const resizeHi = 10
+	expectedEs := make(map[*otlptrace.Span_Link]bool, resizeHi-resizeLo)
+	for i := resizeLo; i < resizeHi; i++ {
+		expectedEs[es.Get(i).orig] = true
```

Comment:
Any particular reason the `orig` field is used as the key of this map and not the `struct` that contains `orig`? It would appear a bit more natural in the test to verify that the actual elements of the slice are correctly resized instead of just one field of the element (even if the element is a single-field `struct`).

Reply:
Done.